### PR TITLE
Signed orders API

### PR DESCRIPTION
**Pull request by atvanguard**
## Why this should be merged
- Introduces Signed orders
- New precompile `jurorv2` that adds functionality to match signed orders with any other order type

Accompanying contracts PR - https://github.com/hubble-exchange/hubble-protocol/pull/214

### CODE REVIEW

**Review by shubhamgoyal42 — APPROVED** [review]
there are some commented log lines which can be removed

File: plugin/evm/orderbook/hubbleutils/signed_orders.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+		PrimaryType: "Order",
+		Domain:      domain,
+		Message:     message,
+	}
+	return EncodeForSigning(typedData)
+}
+
+// Trading API methods
+
+// func (o *SignedOrder) UnmarshalJSON(data []byte) error {
```
[the remark this answers is not in the document]
Comment (atvanguard) [reply plugin/evm/orderbook/hubbleutils/signed_orders.go:129]: Used this earlier, but now I changed encoding from json to abo encoded

File: plugin/evm/orderbook/memory_database.go [hunk 2]
```diff
@@ -79,17 +79,14 @@ const (
 	Execution_Failed
 )
 
-type OrderType uint8
+type OrderType = hu.OrderType
 
 const (
-	Limit OrderType = iota
-	IOC
+	Limit  = hu.Limit
```
[the remark this answers is not in the document]
Comment (atvanguard) [reply plugin/evm/orderbook/memory_database.go:85]: Moved the data structures to hu. but a lot of code was referencing `Limit` directly from this package.

File: plugin/evm/orderbook/config_service.go [hunk 3]
```diff
@@ -102,3 +103,15 @@ func (cs *ConfigService) GetCumulativePremiumFraction(market Market) *big.Int {
...
+
+func (cs *ConfigService) GetSignedOrderStatus(orderHash common.Hash) int64 {
+	return bibliophile.GetSignedOrderStatus(cs.getStateAtCurrentBlock(), orderHash)
+}
+
+func (cs *ConfigService) IsTradingAuthority(trader, signer common.Address) bool {
+	return bibliophile.IsTradingAuthority(cs.getStateAtCurrentBlock(), trader, signer)
+}
+
+func (cs *ConfigService) GetChainIdAndSignedOrderbookContract() common.Address {
```
[the remark this answers is not in the document]
Comment (atvanguard) [reply plugin/evm/orderbook/config_service.go]: done.

File: plugin/evm/orderbook/trading_apis.go [hunk 4]
```diff
@@ -310,3 +313,81 @@ func (api *TradingAPI) StreamMarketTrades(ctx context.Context, market Market, bl
 
 	return rpcSub, nil
 }
+
+type PlaceOrderResponse struct {
+	Success bool `json:"success"`
+}
+
+func (api *TradingAPI) PostOrder(ctx context.Context, rawOrder string) (PlaceOrderResponse, error) {
```
[the remark this answers is not in the document]
Comment (atvanguard) [reply plugin/evm/orderbook/trading_apis.go]: done.

**Pull request merged**